### ISSUE TITLE
Display window controls are malfunctioning

### ISSUE THREAD

**Issue by ux-git** [sircharlo/meeting-media-manager#3314]
### What happened?

1. Can't switch to the third monitor
2. Can't change the window to "Windowed" (it tries, but then goes back to full screen)

This time, even restarting the app won't help

### To Reproduce

Change the display window controls

### What did you expect to happen?

To follow the user instructions

### Version

v24.11.1

### What platform are you seeing the problem on?

MacOS

### Screenshots

https://github.com/user-attachments/assets/b337be60-dfc0-4334-a302-5f8fbfc41a0f

### Confirmations

- [X] I confirm that I have read the documentation website.
- [X] I confirm that I have searched through the existing issues to make sure my bug report is not a duplicate.
- [X] I confirm that I have searched through the existing discussions to make sure my bug report is not a duplicate.

**Comment by mtdvlpr** [comment]
I think this is a duplicate of #3280.

**Comment by mtdvlpr** [comment]
See https://github.com/sircharlo/meeting-media-manager/issues/3256#issuecomment-2476233978 for a test version that should fix it.

**Comment by ux-git** [comment]
For me, none of the described bug are seem to be present: 

> The fullscreen media window no longer "flees" the main window
> The fullscreen media window moves to the first monitor whenever the main window is moved, even if monitor 3 is selected in the display popup
> The fullscreen media window no longer remembers which monitor it was on during the last run
> The main window refuses to return to the first monitor if it was there during the last run. Instead, the media window goes to the first monitor and the main window goes to the second screen.

**Comment by ux-git** [comment]
@mtdvlpr the window vs full screen works fine now after the latest test build update. But switching to the 3rd monitor does not occur immediately. I have to toggle the window type to windowed, and then back to full screen for the display monitor to be changed.

**Comment by sircharlo** [comment]
This is fixed for me on Windows in the latest official release. Can you try again just to be sure?

**Comment by ux-git** [comment]
@sircharlo the windowed/fullscreen control is fine, but moving between displays still won't work on macOS unless I restart the app or toggle the full screen mode off/on.

**Comment by ux-git** [comment]
@sircharlo it doesn't quite work 😅

https://github.com/user-attachments/assets/67890025-f69c-4749-9a66-7e0400371190

It won't allow to close the app neither. I manage to pause this window-dance, but switching the full screen display monitor worked only once, but afterward not anymore.

**Comment by sircharlo** [comment]
Ok; macOS fullscreen behaves differently than other OS's (of course).
Can you try [this build](https://github.com/sircharlo/meeting-media-manager/actions/runs/11856938094) please?

**Comment by ux-git** [comment]
@sircharlo switching between the full-screen displays now works well.

But now I can't switch between full screen and windowed view. Instead of switching to the windowed mode, it gets looped the same as in the previous comment, but the reason is different this time.

**Comment by sircharlo** [comment]
Ok, let's try [this build](https://github.com/sircharlo/meeting-media-manager/actions/runs/11894284038) then. FWIW, this one works perfectly on a 3-monitor Windows setup.

**Comment by ux-git** [comment]
@sircharlo thanks, for everything works! 

Windows require 2-3 resized to properly fit onto the screen or resize from full screen to the windowed mode, but it's definitely working. Once I performed all the possible target screen and window type actions, the further switches were smoother (not 2-3 jumps per switch, but only 1 or none). This jumpy behaviour would be nice to fix, but if it's not easy it can be like this (before someone else reports the issue 😆 ).

Here is the video. I'm changing a screen or window type only after the previous action is performed. So each jump that you see here is not because of quick mode switching, but after a normal "click once, wait 5s" scenario:


https://github.com/user-attachments/assets/7cbf522b-871b-4454-a531-25862e87a9d8

**Comment by sircharlo** [comment]
> Windows require 2-3 resized to properly fit onto the screen or resize from full screen to the windowed mode

Sorry, what do you mean by the above quote? I want to make sure I understand correctly 😄

**Comment by ux-git** [comment]
@sircharlo I mean that after I switch between "Display 1" and "Display 3", the window switches right away, but in order to properly reside, it's doing this in the "full screen → windowed → full screen → windowed → full screen" sequence as shown on the video above. That's something similar to this bug: https://github.com/sircharlo/meeting-media-manager/issues/3314#issuecomment-2478377276 but it stops automatically after doing so 1-3 times.

So the result is correct, but the way to the result it ragged/jumpy.

The reason could be that I was switching between normal external monitor and the main screen

**Comment by sircharlo** [comment]
> @sircharlo I mean that after I switch between "Display 1" and "Display 3", the window switches right away, but in order to properly reside, it's doing this in the "full screen → windowed → full screen → windowed → full screen" sequence as shown on the video above. That's something similar to this bug: [#3314 (comment)](https://github.com/sircharlo/meeting-media-manager/issues/3314#issuecomment-2478377276) but it stops automatically after doing so 1-3 times.
> 
> So the result is correct, but the way to the result it ragged/jumpy.
> 
> The reason could be that I was switching between normal external monitor and the main screen which have different aspect ratio

I see; in that case, I believe that I'll leave it as is for now. Not having a Mac to test it out is brutal 😆 If ever someone has a Mac and wants to try out various scenarios to see where the quirk lies, I'll be more than happy. 

For now though, I think I'll settle with "looks and works great on Windows" and  "works on macOS even though it looks weird during the transition".
